weather-pwa/src/app/app.module.ts

```diff
@@ -8,16 +8,26 @@ import { StatusBar } from '@ionic-native/status-bar/ngx';
 
 import { AppComponent } from './app.component';
 import { AppRoutingModule } from './app-routing.module';
+import { HttpClientModule } from '@angular/common/http';
+import { ForecastManagerService } from './managers';
+import { HttpService } from './services';
+
+const SERVICES = [
+  HttpService,
+  ForecastManagerService,
+];
 
 @NgModule({
   declarations: [AppComponent],
   entryComponents: [],
-  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule],
+  imports: [BrowserModule, IonicModule.forRoot(), AppRoutingModule, HttpClientModule],
   providers: [
+    ...SERVICES,
```

Comment:
What are this 3 magic points?

Reply:
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Operators/Spread_operator